src/NorthwindWebsite/Filters/LoggingActionFilter.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.AspNetCore.Mvc.Filters;
+using NorthwindWebsite.Core.ApplicationSettings;
+
+namespace NorthwindWebsite.Filters;
+
+public class LoggingFilter : IActionFilter
+{
+    private readonly ILogger<LoggingFilter> _logger;
+    private readonly AppSettings _appSettings;
+    private readonly bool _isLoggingEnabled;
+
+    public LoggingFilter(
+        ILogger<LoggingFilter> logger,
+        AppSettings appSettings)
+    {
+        _logger = logger;
+        _appSettings = appSettings;
+        _isLoggingEnabled = appSettings.SerilogConfiguration.ActionsLoggingEnabled;
+    }
+
+    public void OnActionExecuting(ActionExecutingContext context)
+    {
+        if (!_isLoggingEnabled)
+        {
+            return;
+        }
+
+        _logger.LogWarning(string.Format("Action '{0}' started...", context.ActionDescriptor.DisplayName));
+        _logger.LogWarning(string.Format("Method: {0}", context.HttpContext.Request.Method));
+        _logger.LogWarning(string.Format("Path: {0}", context.HttpContext.Request.Path));
+        _logger.LogWarning("Request headers:");
```

Comment:
I think this message should be also moved to `LogHeaders` and after you iterate your headers collection just put this log before first message, you will not have to duplicate it on `45` row